ops/charms/node_base.py

```diff
@@ -2,37 +2,44 @@
 
 import json
 import logging
-import time
+import ops
+from subprocess import run
 from os import PathLike
 from pathlib import Path
-from subprocess import run
+
+import time
 from typing import Union, List, Mapping, Optional, Protocol, Tuple
 
-from ops import Model, Object, CharmMeta, StoredState
+try:
+    from typing import Annotated, TypeAlias
+
+    PositiveInt: TypeAlias = Annotated[int, lambda x: x > 0]
+except ImportError:
+    # if Annotated is not available, just use int
+    PositiveInt = int
 
 log = logging.getLogger(__name__)
 RUN_RETRIES = 180
 
 
-class Charm(Protocol):
+class CharmImpl(Protocol):
```

Comment:
This looks quite Java'ish. I don't see any explicit naming conventions in PEP 544, but I can infer something similar to: https://github.com/python/typeshed/blob/main/CONTRIBUTING.md#naming
Minor nitpick